src/transformers/modeling_utils.py

```diff
@@ -237,9 +237,9 @@ def invert_attention_mask(self, encoder_attention_mask: Tensor) -> Tensor:
         encoder_extended_attention_mask = encoder_extended_attention_mask.to(dtype=self.dtype)  # fp16 compatibility
 
         if self.dtype == torch.float16:
-            encoder_extended_attention_mask = (1.0 - encoder_extended_attention_mask) * -1e4
+            encoder_extended_attention_mask = (1.0 - encoder_extended_attention_mask) * -float("inf")
```

Comment:
This breaks `float16` - we should go with https://pytorch.org/docs/stable/type_info.html `max` here.